src/main/java/fr/inria/gforge/spoon/configuration/AbstractSpoonConfigurationBuilder.java

```diff
@@ -131,7 +131,11 @@ public SpoonConfigurationBuilder addSourceClasspath() throws SpoonMavenPluginExc
 	@Override
 	public SpoonConfigurationBuilder addNoClasspath() {
 		if (spoon.isNoClasspath()) {
-			parameters.add("-x");
+			parameters.add("--cpmode");
+			parameters.add("NOCLASSPATH");
+		} else {
+			parameters.add("--cpmode");
+			parameters.add("FULLCLASSPATH");
```

Comment:
Well you know better than me the parameters for this feature so I trust you here but I definitely think we should refactor it to not rely anymore on the command line arguments: it's really fragile in case of evolution.

Reply:
agree, but we don't have a better solution for now